SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp

```diff
@@ -1223,49 +1222,55 @@ void FSpatialGDKEditorToolbarModule::GenerateSchema(bool bFullScan)
 {
 	LocalDeploymentManager->SetRedeployRequired();
 
-	bSchemaBuildError = false;
+	const bool bFullScanRequired = SpatialGDKEditorInstance->FullScanRequired();
 
-	if (SpatialGDKEditorInstance->FullScanRequired())
+	FSpatialGDKEditor::ESchemaGenerationMethod GenerationMethod;
+	FString OnTaskStartMessage;
+	FString OnTaskCompleteMessage;
+	FString OnTaskFailMessage;
+	if (bFullScanRequired || bFullScan)
 	{
-		OnShowTaskStartNotification("Initial Schema Generation");
-
-		if (SpatialGDKEditorInstance->GenerateSchema(FSpatialGDKEditor::FullAssetScan))
-		{
-			OnShowSuccessNotification("Initial Schema Generation completed!");
-		}
-		else
-		{
-			OnShowFailedNotification("Initial Schema Generation failed");
-			bSchemaBuildError = true;
-		}
+		GenerationMethod = FSpatialGDKEditor::FullAssetScan;
+		const TCHAR* RequiredStr = bFullScanRequired ? TEXT(" required") : TEXT("");
+		OnTaskStartMessage = FString::Printf(TEXT("Generating schema (full scan%s)"), RequiredStr);
+		OnTaskCompleteMessage = TEXT("Full schema generation complete");
+		OnTaskFailMessage = TEXT("Full schema generation failed");
 	}
-	else if (bFullScan)
+	else
 	{
-		OnShowTaskStartNotification("Generating Schema (Full)");
+		GenerationMethod = FSpatialGDKEditor::InMemoryAsset;
+		OnTaskStartMessage = TEXT("Generating schema (incremental)");
+		OnTaskCompleteMessage = TEXT("Incremental schema generation completed!");
+		OnTaskFailMessage = TEXT("Incremental schema generation failed");
+	}
+
+	OnShowTaskStartNotification(OnTaskStartMessage);
+	TSharedFuture<bool> SchemaResult = SpatialGDKEditorInstance->GenerateSchema(GenerationMethod).Share();
+	GenerateSchemaResult(SchemaResult, OnTaskCompleteMessage, OnTaskFailMessage);
+}
 
-		if (SpatialGDKEditorInstance->GenerateSchema(FSpatialGDKEditor::FullAssetScan))
+void FSpatialGDKEditorToolbarModule::GenerateSchemaResult(TSharedFuture<bool> SchemaResult, const FString& OnTaskCompleteMessage,
+														  const FString& OnTaskFaliMessage)
+{
+	if (SchemaResult.IsReady())
+	{
+		bool bResult = SchemaResult.Get();
+		if (bResult)
 		{
-			OnShowSuccessNotification("Full Schema Generation completed!");
+			OnShowSuccessNotification(OnTaskCompleteMessage);
 		}
 		else
 		{
-			OnShowFailedNotification("Full Schema Generation failed");
-			bSchemaBuildError = true;
+			OnShowFailedNotification(OnTaskFaliMessage);
 		}
 	}
 	else
 	{
-		OnShowTaskStartNotification("Generating Schema (Incremental)");
-
-		if (SpatialGDKEditorInstance->GenerateSchema(FSpatialGDKEditor::InMemoryAsset))
-		{
-			OnShowSuccessNotification("Incremental Schema Generation completed!");
-		}
-		else
-		{
-			OnShowFailedNotification("Incremental Schema Generation failed");
-			bSchemaBuildError = true;
-		}
+		/* Wait for the schema result to become available. */
+		Async(EAsyncExecution::Thread, [this, SchemaResult, OnTaskCompleteMessage, OnTaskFaliMessage]() {
```

Comment:
I think lambdas default capture by value. Maybe we can capture refs to the strings?

Reply:
Changed the interface to FString and used MoveTemp in all instances - the initial string will be moved to the final async task!